pkg/actions/create/nodegroup.go

```diff
@@ -0,0 +1,307 @@
+package create
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/kris-nova/logger"
+	"github.com/pkg/errors"
+
+	defaultaddons "github.com/weaveworks/eksctl/pkg/addons/default"
+	"github.com/weaveworks/eksctl/pkg/cfn/manager"
+	"github.com/weaveworks/eksctl/pkg/ctl/cmdutils/filter"
+	"github.com/weaveworks/eksctl/pkg/eks"
+	"github.com/weaveworks/eksctl/pkg/kubernetes"
+	"github.com/weaveworks/eksctl/pkg/utils"
+	"github.com/weaveworks/eksctl/pkg/vpc"
+
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/authconfigmap"
+	"github.com/weaveworks/eksctl/pkg/ctl/cmdutils"
+	"github.com/weaveworks/eksctl/pkg/printers"
+)
+
+// NodeGroupOptions controls specific steps of node group creation
+type NodeGroupOptions struct {
+	UpdateAuthConfigMap       bool
+	InstallNeuronDevicePlugin bool
+}
+
+// NodeGroup holds everything needed to create node groups
+type NodeGroup struct {
+	cfg             api.ClusterConfig
+	nodegroupFilter filter.NodeGroupFilter
+	clusterProvider eks.ClusterProvider
+	options         NodeGroupOptions
+}
+
+// NewNodeGroup returns a NodeGroup create action
+func NewNodeGroup(
+	cfg api.ClusterConfig, nodegroupFilter filter.NodeGroupFilter, clusterProvider eks.ClusterProvider, options NodeGroupOptions,
+) (*NodeGroup, error) {
+	return &NodeGroup{
+		cfg,
+		nodegroupFilter,
+		clusterProvider,
+		options,
+	}, nil
+}
+
+func (c *NodeGroup) Create() error {
+	cfg := &c.cfg
+	meta := cfg.Metadata
+
+	printer := printers.NewJSONPrinter()
+
+	ctl := &c.clusterProvider
+
+	cmdutils.LogRegionAndVersionInfo(meta)
+
+	if err := ctl.CheckAuth(); err != nil {
+		return err
+	}
+
+	if ok, err := ctl.CanOperate(cfg); !ok {
+		return err
+	}
+
```

Comment:
I've seen these bits of logic all over the place, not suggesting as part of this PR but we should consider trying to move this into some shared step that occurs before a commands logic gets invoked. Something analogous to a [middleware](https://github.com/gorilla/mux#middleware)

Reply:
yeah for sure, there's a lot of duplication here